packages/ai/core/tool/mcp/mcp-client.test.ts

```diff
@@ -297,4 +297,14 @@ describe('MCPClient', () => {
       content: [{ type: 'text', text: 'Mock tool call result' }],
     });
   });
+
+  it('should throw if strict mode is enabled and the server does not support tools', async () => {
+    client = await createMCPClient({
+      transport: { type: 'sse', url: 'https://example.com/sse' },
+      strictMode: true,
+    });
+    // Access private property serverCapabilities:
+    (client as any).serverCapabilities = {};
```

Comment:
is this approach necessary or can we mock a transport that responds with this correctly?

Reply:
(I would prefer testing against a scenario, and not set up failure values)

Reply:
on it!